1.9 release docs (#329)
kubefirst/kubefirst: docs/kubefirst/gitlab/argocd.md

# Argo CD

> Argo CD is a GitOps oriented continuous delivery tool for kubernetes. We use it to manage all of our applications across our kubernetes clusters.

Reviewer: ```suggestion
Argo CD is a GitOps oriented continuous delivery tool for Kubernetes. We use it to manage all of our applications across our Kubernetes clusters.
```

> Argo CD is really great at providing a straightforward mechanism for you to manage helm charts, their versions, their configuration overrides, and whether their state is in sync with what's running on your clusters.

Reviewer: ```suggestion
Argo CD is really great at providing a straightforward mechanism for you to manage Helm charts, their versions, their configuration overrides, and whether their state is in sync with what is running on your clusters.
```

![](../../img/kubefirst/argocd/argocd-apps.png)

> ## Managing apps in ArgoCD

Reviewer: ```suggestion
## Managing Apps in Argo CD
```

> Our kubernetes apps can be found in our `gitops` repository at path `/registry`

Reviewer: ```suggestion
Our Kubernetes apps can be found in our `gitops` repository at path `/registry`.
```

![](../../img/kubefirst/gitops/registry.png)

> Each of these yaml files will include details regarding the application's source, destination, and helm configuration overrides.

Reviewer: ```suggestion
Each of these yaml files will include details regarding the application's source, destination, and Helm configuration overrides.
```

## Applying changes in ArgoCD

> Once you change the desired state of the app in the `gitops` repository `main` branch, the registered app-of-apps will sync, and any apps that need adjustment will automatically sync with the state that's in git. 

Reviewer: ```suggestion
Once you change the desired state of the app in the `gitops` repository `main` branch, the registered app-of-apps will sync, and any apps that need adjustment will automatically sync with the state that is in Git. 
```

> ## Adding applications to ArgoCD

Reviewer: ```suggestion
## Adding Applications to Argo CD
```

> The registry of our Argo CD apps are all in the `/registry` root directory. Any application added here will be added to argocd. An app can also be an app-of-apps if you require multiple components to be deliveres as a suite. You'll see plenty of examples to follow in the `/registry` directory.

Reviewer: ```suggestion
The registry of our Argo CD apps are all in the `/registry` root directory. Any application added here will be added to Argo CD. An app can also be an app-of-apps if you require multiple components to be delivered as a suite. You'll see plenty of examples to follow in the `/registry` directory.
```
